src/nets/e2e_asr_attctc_th.py

```diff
@@ -2194,3 +2149,60 @@ def forward(self, xs, ilens):
         xs = [xs[i, :ilens[i]] for i in range(len(ilens))]
         xs = pad_list(xs, 0.0)
         return xs, ilens
+
+
+class BRNNP(torch.nn.Module):
+    def __init__(self, cell_type, idim, elayers, cdim, hdim, subsample, dropout):
+        super(BRNNP, self).__init__()
+        for i in six.moves.range(elayers):
+            if i == 0:
+                inputdim = idim
+            else:
+                inputdim = hdim
+
+            if cell_type == 'blstmp':
+                setattr(self, "birnn%d" % i, torch.nn.LSTM(inputdim, cdim, dropout=dropout,
+                                                            num_layers=1, bidirectional=True, batch_first=True))
+            elif cell_type == 'bgrup':
+                setattr(self, "birnn%d" % i, torch.nn.GRU(inputdim, cdim, dropout=dropout,
+                                                           num_layers=1, bidirectional=True, batch_first=True))
+            else:
+                logging.error(
+                    "Error: specified rnn type, {} , is not yet supported. Please choose {} or {}".format(etype, 'blstmp', 'bgrup'))
+                sys.exit()
+
+            # bottleneck layer to merge
+            setattr(self, "bt%d" % i, torch.nn.Linear(2 * cdim, hdim))
+
+        self.elayers = elayers
+        self.cdim = cdim
+        self.subsample = subsample
+
+    def forward(self, xpad, ilens):
+        '''BRNNP forward
+
+        :param xs:
+        :param ilens:
+        :return:
+        '''
+        # logging.info(self.__class__.__name__ + ' input lengths: ' + str(ilens))
+        for layer in six.moves.range(self.elayers):
+            xpack = pack_padded_sequence(xpad, ilens, batch_first=True)
+            bigru = getattr(self, 'birnn' + str(layer))
```

Comment:
To keep the backward compatibility, can you add one check whether if it has a member of 'bilstm' + str(layer) then use that member instead of 'birnn' + str(layer)? I think this would solve the backward compatibility.

Reply:
I think that simply the fact of changing the class name will break backwards compatibility when loading pretrained weights